packages/js/src/cache/notifications-cache.ts

```diff
@@ -0,0 +1,253 @@
+  return JSON.stringify({ tags, read, archived, limit, offset, after });
+};
+
+const getFilterKey = ({ tags, read, archived }: Pick<ListNotificationsArgs, 'tags' | 'read' | 'archived'>): string => {
+  return JSON.stringify({ tags, read, archived });
+};
+
+const getFilter = (key: string): NotificationFilter => {
+  return JSON.parse(key);
+};
+
+// these events should update the notification in the cache
+const notificationEvents: NotificationEvents[] = [
+  'notification.read.pending',
+  'notification.read.resolved',
+  'notification.unread.pending',
+  'notification.unread.resolved',
+  'notification.complete_action.pending',
+  'notification.complete_action.resolved',
+  'notification.revert_action.pending',
+  'notification.revert_action.resolved',
+];
+
+// these events should remove the notification from the cache
+const notificationEventsThatRemove: NotificationEvents[] = [
+  'notification.archive.pending',
+  'notification.unarchive.pending',
+];
+
+// these events should update the notifications in the cache
+const notificationsEvents: NotificationsEvents[] = [
+  'notifications.read_all.pending',
+  'notifications.read_all.resolved',
+];
+
+// these events should remove the notifications from the cache
+const notificationsEventsThatRemove: NotificationsEvents[] = [
+  'notifications.archive_all.pending',
+  'notifications.archive_all_read.pending',
+];
+
+export class NotificationsCache {
+  #emitter: NovuEventEmitter;
+  /**
+   * The key is the stringified notifications filter, the values are the paginated notifications.
+   */
+  #cache: Cache<ListNotificationsResponse>;
+
+  constructor() {
+    this.#emitter = NovuEventEmitter.getInstance();
+    notificationEvents.forEach((event) => {
+      this.#emitter.on(event, this.handleNotificationEvent());
+    });
+    notificationEventsThatRemove.forEach((event) => {
+      this.#emitter.on(event, this.handleNotificationEvent({ remove: true }));
+    });
+    notificationsEvents.forEach((event) => {
+      this.#emitter.on(event, this.handleNotificationsEvent());
+    });
+    notificationsEventsThatRemove.forEach((event) => {
+      this.#emitter.on(event, this.handleNotificationsEvent({ remove: true }));
+    });
+    this.#cache = new InMemoryCache();
+  }
+
+  private updateNotificationInCache = (key: string, data: Notification): boolean => {
+    const notificationsResponse = this.#cache.get(key);
+    if (!notificationsResponse) {
+      return false;
+    }
+
+    const index = notificationsResponse.notifications.findIndex((el) => el.id === data.id);
+    if (index === -1) {
+      return false;
+    }
+
+    const updatedNotifications = [...notificationsResponse.notifications];
+    updatedNotifications[index] = data;
+
+    this.#cache.set(key, { ...notificationsResponse, notifications: updatedNotifications });
+
+    return true;
+  };
+
+  private removeNotificationInCache = (key: string, data: Notification): boolean => {
```

Comment:
```suggestion
  private removeNotificationFromCache = (key: string, data: Notification): boolean => {
```

Or just `removeNotification`